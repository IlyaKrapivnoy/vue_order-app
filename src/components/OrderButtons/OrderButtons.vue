<template>
  <div class="order">
    <button
      v-for="(term, index) in orderTerms"
      :key="index"
      @click="handleClick(term)"
      :class="{ active: term === selectedTerm }"
    >
      Order by {{ term }}
    </button>
  </div>
</template>

<script setup lang="ts">
  import { defineEmits, ref } from 'vue';
  import OrderTerm from '@/types/OderTerm';

  const orderTerms: OrderTerm[] = ['location', 'title', 'salary'];
  const selectedTerm = ref<OrderTerm>('title');

  const emit = defineEmits(['onOrderClick']);

  const handleClick = (term: OrderTerm) => {
    selectedTerm.value = term;
    emit('onOrderClick', term);
  };
</script>

<style scoped>
  .order {
    margin-top: 20px;
  }

  .active {
    background-color: #1195c9;
    color: white;
  }
</style>
