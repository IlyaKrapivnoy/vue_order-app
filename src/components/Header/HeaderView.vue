<template>
  <header>
    <img src="../../assets/logo.webp" alt="site logo" />
    <OrderButtons @onOrderClick="handleOrderClick" />
  </header>
</template>

<script setup lang="ts">
  import { defineProps } from 'vue';

  import OrderButtons from '@/components/OrderButtons/OrderButtons.vue';
  import OrderTerm from '@/types/OderTerm';

  const props = defineProps({
    handleClick: {
      required: true,
      type: Function,
    },
  });

  const handleOrderClick = (term: OrderTerm) => {
    if (typeof props.handleClick === 'function') {
      props.handleClick(term);
    }
  };
</script>

<style scoped>
  header {
    text-align: center;
  }

  header img {
    width: 300px;
    margin: 40px 0 20px;
  }
</style>
