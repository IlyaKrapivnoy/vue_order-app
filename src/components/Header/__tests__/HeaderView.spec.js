import { mount } from '@vue/test-utils';
import HeaderView from '@/components/Header/HeaderView.vue';
import OrderButtons from '@/components/OrderButtons/OrderButtons.vue';

describe('HeaderView.vue', () => {
  it('matches the snapshot', () => {
    const wrapper = mount(HeaderView);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders the site logo', () => {
    const wrapper = mount(HeaderView);
    expect(wrapper.find('img').exists()).toBe(true);
  });

  it('renders the OrderButtons component', () => {
    const wrapper = mount(HeaderView, {
      global: {
        components: {
          OrderButtons,
        },
      },
    });
    expect(wrapper.findComponent(OrderButtons).exists()).toBe(true);
  });
});
