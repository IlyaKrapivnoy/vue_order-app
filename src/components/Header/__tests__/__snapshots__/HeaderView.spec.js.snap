// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderView.vue matches the snapshot 1`] = `
<header><img alt="site logo">
  <div class="order"><button class=""> Order by location</button><button class="active"> Order by title</button><button class=""> Order by salary</button></div>
</header>
`;
